src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/boss_argent_challenge.cpp

```diff
@@ -104,512 +158,931 @@ enum Spells
             }
 
-            if (!bHealth && !HealthAbovePct(25))
-            {
-                me->InterruptNonMeleeSpells(true);
-                DoCastAOE(SPELL_HOLY_NOVA, false);
-                DoCast(me, SPELL_SHIELD);
-                DoCastAOE(SPELL_SUMMON_MEMORY, false);
-                DoCastAOE(SPELL_CONFESS, false);
+        private:
+            InstanceScript* _instance;
+            EventMap _events;
 
-                bHealth = true;
-            }
+            ObjectGuid _fountainGuid;
 
-            DoMeleeAttackIfReady();
-        }
+            uint8 _waypointId;
+            bool _shielded;
+        };
 
-        void JustSummoned(Creature* summon) override
+        CreatureAI* GetAI(Creature* creature) const override
         {
-            MemoryGUID = summon->GetGUID();
+            return GetTrialOfChampionAI<npc_argent_soldierAI>(creature);
         }
-    };
-
-    CreatureAI* GetAI(Creature* creature) const override
-    {
-        return GetTrialOfTheChampionAI<boss_paletressAI>(creature);
-    }
 };
 
-class npc_memory : public CreatureScript
+class npc_fountain_of_light : public CreatureScript
 {
-public:
-    npc_memory() : CreatureScript("npc_memory") { }
+    public:
+        npc_fountain_of_light() : CreatureScript("npc_fountain_of_light") { }
 
-    struct npc_memoryAI : public ScriptedAI
-    {
-        npc_memoryAI(Creature* creature) : ScriptedAI(creature)
+        struct npc_fountain_of_lightAI : public ScriptedAI
         {
-            Initialize();
-        }
+            npc_fountain_of_lightAI(Creature* creature) : ScriptedAI(creature)
+            {
+                me->SetReactState(REACT_PASSIVE);
+                SetCombatMovement(false);
+            }
 
-        void Initialize()
-        {
-            uiOldWoundsTimer = 12000;
-            uiShadowPastTimer = 5000;
-            uiWakingNightmare = 7000;
-        }
+            void JustDied(Unit* /*killer*/) override
+            {
+                me->DespawnOrUnsummon();
+            }
+
+            void AttackStart(Unit* /*who*/) override { }
 
-        uint32 uiOldWoundsTimer;
-        uint32 uiShadowPastTimer;
-        uint32 uiWakingNightmare;
+            void UpdateAI(uint32 /*diff*/) override { }
+        };
 
-        void Reset() override
+        CreatureAI* GetAI(Creature* creature) const override
         {
-            Initialize();
+            return GetTrialOfChampionAI<npc_fountain_of_lightAI>(creature);
         }
+};
 
-        void UpdateAI(uint32 uiDiff) override
+class npc_memory : public CreatureScript
+{
+    public:
+        npc_memory() : CreatureScript("npc_memory") { }
+
+        struct npc_memoryAI : public ScriptedAI
         {
-            if (!UpdateVictim())
-                return;
+            npc_memoryAI(Creature* creature) : ScriptedAI(creature)
+            {
+                Initialize();
+				instance = creature->GetInstanceScript();
+                me->SetReactState(REACT_PASSIVE);
```

Comment:
Just passive, but still attackable? Should there be immunity flags here?

Reply:
iirc, the "image" would spawn, and would be attackable while "spawning", but would not react until it entered the aggressive event, so this is correct

Reply:
Actually, reviewing some old footage of the fight, the memory would be attackable, and would receive debuffs, however it would not take any damage until it entered the aggressive event. So yeah, should not take any damage while in the passive state.

Reply:
Ummm isnt all the party under Confess effect here? (stunned)